docs/blog/2018-12-17-turning-the-static-dynamic/index.md

```diff
@@ -0,0 +1,405 @@
+
+Just like [every magic act has a pledge, a turn, and a prestige](<https://en.wikipedia.org/wiki/The_Prestige_(film)>), I have one last tidbit for you. [Nothing on the client-side is safe](https://stackoverflow.com/questions/50277192/react-security-concerns-restricted-pages-in-app), and although you can send along Netlify Identity user id's to your Netlify Function endpoints for authenticated access from your Gatsby App (for example in the body of a POST request), you'll never be truly sure if that flow is secure either from malicious users or snooping.
+
+The best way to do authenticated actions inside serverless functions is to do it from inside the context of the function itself. Fortunately, [Netlify Identity and Functions work seamlessly together](https://www.netlify.com/docs/functions/#identity-and-functions). All you have to do is to send along the user's [JWT](https://jwt.io/) when hitting your endpoint:
+
+```js
+// in your gatsby app
+const user = getUser()
+fetch("/.netlify/functions/auth-hello", {
+  headers: {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+    Authorization: "Bearer " + user.token.access_token, // like this
+  },
+}).then(/* etc */)
+```
+
+And then inside a lambda function, you can now access the `user` object:
+
+```js
+// more info: https://www.netlify.com/docs/functions/#identity-and-functions
+
+// Note that `netlify-lambda` only locally emulates Netlify Functions, while `netlify-identity-widget` interacts with a real Netlify Identity instance. This means that `netlify-lambda` doesn't support Netlify Functions + Netlify Identity integration.
+
+export function handler(event, context, callback) {
+  if (context.clientContext) {
+    const {
+      user, // actual user info you can use for your serverless functions
+    } = context.clientContext
+    callback(null, {
+      statusCode: 200,
+      body: JSON.stringify({
+        msg: "auth-hello: " + Math.round(Math.random() * 10),
+        user,
+      }),
+    })
+  } else {
+    console.log(`
+    Note that netlify-lambda only locally emulates Netlify Functions, 
+    while netlify-identity-widget interacts with a real Netlify Identity instance. 
+    This means that netlify-lambda doesn't support Netlify Functions + Netlify Identity integration.
+    `)
+    callback(null, {
+      statusCode: 200,
+      body: JSON.stringify({
+        msg:
+          "auth-hello - no authentication detected. Note that netlify-lambda doesnt locally emulate Netlify Identity.",
+      }),
+    })
+  }
+}
+```
+
+## Gatsby - Perfect for your next Hackathon
+
+It's 5 steps each to turn your static Gatsby sites into dynamic, authenticated, fully serverless apps with Netlify's free tools. This makes Gatsby a perfect tool for your next app. If you're at a hackathon, short on time, or just like to see a full working demo, check any of the following links.
+
+- **Code:** https://github.com/sw-yx/jamstack-hackathon-starter
```

Comment:
I can't decide if putting these at the beginning would be awesome or just stop people from reading. Something to consider

Reply:
i put it at the end because i think the goal is “teach them to fish” rather than push my starter on to them... yknow?

Reply:
Yeah, I think I agree. It's better that way!